File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/Java/Exercicio17.java
/*
17 - Faça uma função para criptografar uma frase digitada pelo usuário.
A criptografia deverá substituir todas as vogais por *. Utilize a função no programa principal.
A frase poderá ter, no máximo, 50 caracteres 
*/

import java.util.Scanner;

public class exercicio17 {
    public static void criptografia(String frase){
        String[] nulos = {"A", "E", "I", "O", "U", "a", "e", "i", "o", "u"};
        for (String nulo : nulos) { 
            frase = frase.replaceAll(nulo, "*");
        }
        
        System.out.println(frase);
    }
    
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        System.out.println("Digite uma frase: ");
        String frase = teclado.nextLine();
       
        criptografia(frase);
    }
}

File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/C/Exercicio 17.c
/*
17 - Faça uma função para criptografar uma frase digitada pelo usuário.
A criptografia deverá substituir todas as vogais por *. Utilize a função no programa principal.
A frase poderá ter, no máximo, 50 caracteres 
*/

#include<stdio.h>
#include<string.h>
#include<locale.h>

void criptografia (char vogal[]);

int main ()
{
    char vogal[50];
    setlocale(LC_ALL, "Portuguese");
    printf ("Digite uma frase: ");
    scanf (" %[^\n]",vogal);

    criptografia (vogal);
    printf ("\n %s \n",vogal);
    return 0;
}

void criptografia (char vogal[])
{
    for (int i=0; i<=50; i++){
        if (vogal[i]=='a' || vogal[i]=='e' || vogal[i]=='i' || vogal[i]=='o' || vogal[i]=='u' || vogal[i]=='A' || vogal[i]=='E' || vogal[i]=='I' || vogal[i]=='O' || vogal[i]=='U'){
            vogal[i] = '*';
            i++;
        }
    }
    return ;
}
